tests/typecheck/conf/test_settings.yml

```diff
@@ -1,22 +1,34 @@
 -   case: test_setting_circular_import
     main: |
         from myapp import lib
-    custom_settings: |
-        from myapp.lib import function_returning_int
-
-        IMMEDIATE_VALUE = 123
-        CIRCULAR_WITHOUT_HINT = function_returning_int()
-        CIRCULAR_WITH_HINT: int = function_returning_int()
+    mypy_config: |
+        [mypy.plugins.django-stubs]
+        django_settings_module = myapp.settings
     files:
         -   path: myapp/__init__.py
+        -   path: myapp/settings.py
+            content: |
+                from myapp.lib import function_returning_int, const_with_circular_import
+
+                IMMEDIATE_VALUE = 123
+                CIRCULAR_WITH_HINT: int = function_returning_int()
+                CIRCULAR_WITHOUT_HINT_FUNCTION = function_returning_int()
+                CIRCULAR_WITHOUT_HINT_CONST = const_with_circular_import  # E: Cannot determine type of "const_with_circular_import"  [has-type]
```

Comment:
Well, I managed to coax a different error out of mypy.

Reply:
Interesting, I think that's totally unresolvable for mypy since there's no type hint involved at all for the type checker

Reply:
Oh, right, it used `settings.IMMEDIATE_VALUE`. Then there's actually a chance for it to figure it out